### ISSUE TITLE
Setting objectStorageConfig in ThanosRuler produces invalid volume for StatefulSet

### ISSUE THREAD

**Issue by zbialik** [prometheus-operator/prometheus-operator#5256]
**What happened?**

Define a `ThanosRuler` like so:

```
apiVersion: monitoring.coreos.com/v1
kind: ThanosRuler
metadata:
  name: main
spec:
  objectStorageConfig:
    key: thanos.yaml
    name: thanos-objectstorage
```

Watched a `StatefulSet` fail to provision with error:
`create Pod thanos-ruler-main-0 in StatefulSet thanos-ruler-main failed error: Pod "thanos-ruler-main-0" is invalid: [spec.volumes[0].secret.items[0].path: Required value, spec.containers[0].volumeMounts[0].name: Not found: "objstorage-config"]`

The `StatefulSet` generated is shown below (cleaned):

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: thanos-ruler-main
spec:
  selector:
    matchLabels:
      thanos-ruler: main
  template:
    metadata:
      labels:
        thanos-ruler: main
    spec:
      containers:
        - args:
            - rule
            - --data-dir=/thanos/data
            - --eval-interval=15s
            - --tsdb.retention=24h
            - --label=thanos_ruler_replica="$(POD_NAME)"
            - --alert.label-drop=thanos_ruler_replica
            - --objstore.config-file=/etc/thanos/config/objstorage-config.yaml
          image: thanos/thanos:v0.25.2
          name: thanos-ruler
          volumeMounts:
            - mountPath: /etc/thanos/config/objstorage-config.yaml
              name: objstorage-config
      volumes:
        - name: objstorage-config
          secret:
            defaultMode: 420
            items:
              - key: thanos.yaml
                path: ""
            secretName: thanos-objectstorage
```

**Did you expect to see something different?**

The `Prometheus` CRDs generate `StatefulSets` with the objstorage-config as an environment variable (using `envFrom`), rather than a volume. 

See the snippet of the `thanos-sidecar` container config in a `Prometheus` generated `StatefulSet` below:

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  labels:
    operator.prometheus.io/name: main
  name: prometheus-main
spec:
  selector:
    matchLabels:
      prometheus: main
  template:
    metadata:
      labels:
        prometheus: main
    spec:
      containers:
        - args:
            - sidecar
            - --prometheus.url=http://localhost:9090/
            - --grpc-address=:10901
            - --http-address=:10902
            - --objstore.config=$(OBJSTORE_CONFIG)
            - --tsdb.path=/prometheus
            - --log.level=info
          env:
            - name: OBJSTORE_CONFIG
              valueFrom:
                secretKeyRef:
                  key: thanos.yaml
                  name: thanos-objectstorage
          image: thanos/thanos:v0.29.0
          name: thanos-sidecar
```

I think the `ThanosRuler` could do something similar or fix the generated volume config to not set `path: ""` (as I think that is the root issue)

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a valid objstore-config `Secret` 
2. Create the `ThanosRuler` CRD below

```
apiVersion: monitoring.coreos.com/v1
kind: ThanosRuler
metadata:
  name: main
spec:
  objectStorageConfig:
    key: thanos.yaml
    name: thanos-objectstorage
```

3. watch `StatefulSet` fail to provision `Pods`

**Environment**

* Prometheus Operator version: `v0.61.0`
* Kubernetes version information: `v1.23`
* Kubernetes cluster kind: `eks`

**Comment by zbialik** [comment]
**UPDATE**: 
If you also set the `objectStorageConfigFile`, then the `StatefulSet` is generated with the `envFrom` style manifest, similar to what is done for `Prometheus` CRD. This fixes the issue.

However, I don't think it should be required that we set both fields in the `ThanosRuler` CRD to get the `StatefulSet` to generate correctly.

**Comment by simonpasquier** [comment]
This should be fixed by #5183 which has been included in v0.61.1. Can you give it a try?
